packages/api-folders-so-ddb/src/definitions/entities.ts

```diff
@@ -0,0 +1,90 @@
+import { Entity, Table } from "dynamodb-toolbox";
+import { Attributes, ENTITIES } from "../types";
+
+const createEntity = (entityName: string, table: Table, attributes: Attributes) => {
+    return new Entity({
+        table,
+        name: entityName,
+        attributes: {
+            PK: {
+                partitionKey: true
+            },
+            SK: {
+                sortKey: true
+            },
+            GSI1_PK: {
+                type: "string"
+            },
+            GSI1_SK: {
+                type: "string"
+            },
+            TYPE: {
+                type: "string"
+            },
+            ...(attributes || {})
+        }
+    });
+};
+
+export const createFolderEntity = (table: Table, attributes: Attributes = {}) => {
+    return createEntity(ENTITIES.FOLDER, table, {
+        id: {
+            type: "string"
+        },
+        name: {
+            type: "string"
+        },
+        slug: {
+            type: "string"
+        },
+        category: {
+            type: "string"
+        },
+        createdOn: {
+            type: "string"
+        },
+        createdBy: {
+            type: "map"
+        },
+        tenant: {
+            type: "string"
+        },
+        locale: {
+            type: "string"
+        },
```

Comment:
Let's store these into a `data` field, so it's easy to extend via plugins. We made this mistake of having individual fields several times, and it's super difficult to extend. Let's go with the approach which makes it easy to extend, so just put all folder related properties into a `data` map. An example: `packages/api-prerendering-service-so-ddb/src/definitions/render.ts:32`